### DISCUSSION TITLE
manage state of host app and mini app

### DISCUSSION THREAD

**Discussion by HongQuang231** [callstack/repack#511] [Q&A]
how to create global state in host app, global state in mini app, and share state between two app? I want to update state from host app by mini app? Can you help me?

**Comment by jbroma** [comment]
hey @HongQuang231,

If you want to use Context based state management solutions (that have a Provider in a React tree) then it works exactly the same - you place the provider high enough in the component tree and you can access it in the mini-app as well.

When we are talking about a state management solution like `Zustand`, which creates a singleton instance that should then be imported into other files there a few ways to do that:
1. You could create a separate federated module and keep the state management there and import it through Module Federation in both host and mini-app
2. You could create an instance inside of the host app and do something like `global.dataStore = store` and then you can access the store in mini-app that way - in `Zustand` that would be the result of `useStore` function.

**Reply by HongQuang231** [reply]
can i use redux in my project using repack? and you have a project example for this. I want to understand what you mean by clearing a mini project. Thanks
